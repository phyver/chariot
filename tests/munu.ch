data c('x) where C1 : 'x -> c('x)
codata t1 where D1 : t1 -> c(t1)

codata d('x) where D2 : d('x) -> 'x
data t2 where C2 : d(t2) -> t2

val test1 : t1
  | test1 = D1 # C1 $ test1

val test2 : c(t1)
    | test2 = C1 $ D1 # test2

val test3 : t2
  | test3 = C2 $ D2 # test3

val test4 : d(t2)
    | test4 = D2 # test3

:show functions

