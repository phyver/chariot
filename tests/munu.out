>> option verbose is set to 1
>> inductive type c was succesfully defined
>> coinductive type t1 was succesfully defined
>> coinductive type d was succesfully defined
>> inductive type t2 was succesfully defined
>> Typing for test1 successful
>> the definition for test1 is complete
>> the definition test1 is provably correct

>> Typing for test2 successful
>> the definition for test2 is complete
>> the definition test2 is provably correct

>> Typing for test3 successful
>> the definition for test3 is complete
!! warning: the definition test3 is NOT provably correct (weight_bound: 2, depth_bound: 2)

>> Typing for test4 successful
>> the definition for test4 is complete
>> the definition test4 is provably correct

>> functions in environment:
val
   test1 : t1
    | test1 = { D1 = C1¹ test1 }²

val
   test2 : c(t1)
    | test2 = C1¹ { D1 = test2 }²

val
   test3 : t2
    | test3 = C2³ { D2 = test3 }²

val
   test4 : d(t2)
    | test4 = { D2 = test3 }²

